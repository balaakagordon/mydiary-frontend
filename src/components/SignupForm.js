import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signupAction } from '../actions/signupActions';
import { toast } from 'react-toastify';
import history from '../history';
import image from "../images/diary2.jpeg"


export class SignupForm extends Component {
  state = {
    user: {
      name: "",
      email: "",
      password: "",
      confirmpassword: ""
    },
    status: "none"
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.status) {
      if (nextProps.status === 'loading') {
        console.log("SHOW THE LOADER")
      } else if (nextProps.status === 'success') {
        history.push('/');
      } else if (nextProps.status === 'error') {
        this.displayError(nextProps.message)
      }
    }
  }

  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = event => {
    event.preventDefault();
    const userdata = {
        name: this.state.name,
        email: this.state.email,
        password: this.state.password,
        confirmpassword: this.state.confirmpassword
    };
    this.props.signupAction(userdata);
  };

  displayError = message => {
    toast.error(message, {
      position: toast.POSITION.TOP_CENTER,
      hideProgressBar:true
    });
  };

  render() {
    return (
      <div className="container auth-form">
        <section>
          <div className="container">
            <div className="row">
              <div className="col-md-4">
                <form onSubmit={this.handleSubmit} id="login-form">
                  <h2 className="centred">Register</h2>
                  <br />
                  <div className="row">
                    <input
                      className="auth-input"
                      type="text"
                      name="name"
                      id="name"
                      placeholder="Enter your name here"
                      onChange={this.onChange}/>
                  </div>
                  <div className="row">
                    <input
                      className="auth-input"
                      type="text"
                      name="email"
                      id="email"
                      placeholder="Enter email here"
                      onChange={this.onChange}/>
                  </div>
                  <div className="form-group row">
                    <input
                      className="form-control auth-input"
                      type="password"
                      name="password"
                      id="password"
                      placeholder="Enter password here"
                      onChange={this.onChange}/>
                  </div>
                  <div className="form-group row">
                    <input
                      className="form-control auth-input"
                      type="password"
                      name="confirmpassword"
                      id="confirmpassword"
                      placeholder="Confirm password here"
                      onChange={this.onChange}/>
                  </div>
                  <br />
                  <button
                    type="submit"
                    className="auth-submit"
                    id="submit">
                    Submit
                  </button>
                  <br />
                  <p className="small-text">
                  <br />
                    <u>
                      <Link to="/">already have an account?</Link>
                    </u>
                  </p>
                </form>
              </div>
              <div className="col-md-8">
                <div id="login-image">
                <img
                  className="auth-img"
                  src={image}
                  alt="Authentication"
                  height="480px"/>
                </div>  
              </div>
          </div>
        </div>
      </section>
      </div>
    );
  }
}

SignupForm.propTypes = {
  signupAction: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.signup.user,
  message: state.signup.message,
  status: state.signup.status
});

export default withRouter(
  connect(
    mapStateToProps,
    { signupAction }
  )(SignupForm)
);
// export default SignupForm;
