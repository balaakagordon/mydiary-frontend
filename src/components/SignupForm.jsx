import React from 'react';
import { Link } from 'react-router-dom';
import FormInputField from './FormComponents/FormInputField';
import Button from './FormComponents/Button';


const SignupForm = (props) => {
  return (
    <div className="auth-form">
      <div className="loader">{props.showLoader}</div>
      <FormInputField type={'text'}
          title={'First Name'}
          name={'firstName'}
          value={props.user.firstName}
          placeholder={'Enter your first name'}
          handleChange={props.handleFormInput}
          className={'auth-form-input'}
      />
      <FormInputField type={'text'}
          title={'Last Name'}
          name={'lastName'}
          value={props.user.lastName}
          placeholder={'Enter your last name'}
          handleChange={props.handleFormInput}
          className={'auth-form-input'}
      />
      <FormInputField type={'text'}
          title={'Email'}
          name={'email'}
          value={props.user.email}
          placeholder={'Enter your email address'}
          handleChange={props.handleFormInput}
          className={'auth-form-input'}
      />
      <FormInputField type={'password'}
          title={'Password'}
          name={'password'}
          value={props.user.password}
          placeholder={'Enter your password'}
          handleChange={props.handleFormInput}
          className={'auth-form-input'}
      />
      <FormInputField type={'password'}
          title={'Confirm Password'}
          name={'confirmedPassword'}
          value={props.user.confirmPassword}
          placeholder={'Re-enter your password'}
          handleChange={props.handleFormInput}
          className={'auth-form-input'}
      />
      <div className="auth-link">
        <Link to="/">already have an account?</Link>
      </div>
      <div className='row'>
        <div className='form-buttons'>
          <Button
              action={props.handleFormSubmit}
              title={'Submit'}
              className='auth-button auth-submit'
          />
          <Button
              action={props.handleClearForm}
              title={'Cancel'}
              className='auth-button auth-cancel'
          />
        </div>
      </div>
    </div>
  );
}

export default SignupForm;
