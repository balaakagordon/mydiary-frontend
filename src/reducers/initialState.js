export default {
    signup: {
        loading: false,
        message: null,
        status: '',
        token: '',
        errors: null
    },
    login: {
        loading: false,
        message: null,
        status: '',
        token: '',
        errors: null
    },
    getEntries: {
        entriesData: {
            entries: [],
            total: 0,
            currentPage: 1,
            from: null,
            to: null,
            lastPage: null,
            perPage: null,
            firstPageUrl: null,
            lastPageUrl: null,
            prevPageUrl: null,
            nextPageUrl: null
        },
        loading: false,
        message: null,
        status: null,
        errors: null
    },
    newEntry: {
        entry: {
            title: '',
            body: '',
            lastEdited: null
        },
        loading: false,
        status: null,
        message: null,
        errors: null
    },
    updateEntry: {
        entry: {
            title: '',
            body: '',
            lastEdited: null
        },
        loading: false,
        status: null,
        message: null,
        errors: null
    },
    getEntry: {
        entry: {
            id: null,
            author: null,
            title: '',
            body: '',
            lastEdited: null
        },
        loading: false,
        status: null,
        message: null,
        errors: null
    },
    getProfile: {
        profile: {
            id: null,
            firstName: null,
            lastName: null,
            email: null,
            image: null,
            lastUsed: null,
            currentEntries: null,
            allEntries: null,
            notifications: null,
            createdAt: null,
            // selectedFile: null
        },
        loading: false,
        status: null,
        errors: null
    },
    navbar: {
        authenticated: false,
    },
  };