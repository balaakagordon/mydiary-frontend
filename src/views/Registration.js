import React from 'react';
import SignupForm from '../components/SignupForm'
import Header from '../components/Header';


const Registration = () => (
	// <h1>User Registration Page</h1>
	<div>
		<Header />
		<SignupForm />
	</div>
);

export default Registration;