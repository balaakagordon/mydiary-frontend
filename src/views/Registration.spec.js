import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import initialState from '../reducers/initialState';
import SignupForm from '../components/SignupForm';

import { Registration, mapStateToProps } from './Registration';
import { wrap } from 'module';

describe('The Registration Component', () => {
    describe('The container element', () => {
        it('should match the snapshot', () => {
            const tree = mapStateToProps(initialState);
            expect(tree).toMatchSnapshot();
        });

        it('should map the state to props correctly', () => {
            const sampleProps = {
                user: {
                    firstName: 'Da',
                    lastName: 'Di',
                    email: 'e.m@email.com',
                    password: 'Password',
                    confirmedPassword: 'Password'
                },
                loading: false,
                message: null,
                status: '',
                errors: null
            };

            const appState = {
                signup: {
                    ...sampleProps
                }
            };

            const ownProps = {
                user: {}
            };

            const componentState = mapStateToProps(appState, ownProps);
            expect(componentState).toEqual(sampleProps);
        });
    });

    const mockStore = configureStore();
    const store = mockStore(initialState.signup);
    let component;
    beforeEach(() => {
        component = shallow(<Registration store={store} {...initialState.signup} />);
    });
    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render without crashing', () => {
        expect(component).toBeDefined();
    })

    it('should render children', () => {
        expect(component.find(SignupForm)).toBeDefined();
    })

    it('should call componentWillReceiveProps', () => {
        expect(component.find(SignupForm)).toBeDefined ();
    })
});

describe('Registration methods', () => {
    let wrapper;
    let event = {
        preventDefault: jest.fn()
    };

    let sessionStorage = {
        getItem: jest.fn((token) => {
            const result = `user.access.${token}`
            return result;
        })
    }

    let history = {
        push: jest.fn()
    };

    beforeEach(() => {
        wrapper = shallow(<Registration />);
        wrapper.instance().props = {
            userSignup: jest.fn()
        };
        wrapper.instance().handleClearForm = jest.fn();
        wrapper.instance().handleFormInput = jest.fn();
        wrapper.instance().handleLoader = jest.fn();
        wrapper.update();
    })
    it("componentWillReceiveProps calls handleSuccess after receiving success props", () => {
        wrapper.instance().handleSuccess = jest.fn();
        wrapper.update();
        const nextProps = {
            status: "success"
        };
        wrapper.setProps(nextProps);
        expect(wrapper.instance().handleSuccess).toBeCalled();
    });

    it("componentWillReceiveProps calls handleErrors after receiving error props", () => {
        wrapper.instance().handleErrors = jest.fn();
        wrapper.instance();
        const nextProps = {
            status: "error"
        };
        wrapper.setProps(nextProps);
        expect(wrapper.instance().handleErrors).toBeCalled();
    });

    it("handleLoader is called when loading props are true", () => {
        const nextProps = {
            loading: true
        };
        wrapper.setProps(nextProps);
        expect(wrapper.instance().handleLoader).toBeCalled();
    });

    it("calls userSignup when form is submitted", () => {
        wrapper.instance().handleFormSubmit(event);
        expect(wrapper.instance().props.userSignup).toBeCalled();
    });

    it("calls redirects to home page handleSuccess is called", () => {
        wrapper.instance().handleSuccess('msg');
        // expect(wrapper.instance().props.userSignup).toBeCalled();
    });
})