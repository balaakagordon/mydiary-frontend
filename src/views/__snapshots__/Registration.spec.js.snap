// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Registration Component The container element should match the snapshot 1`] = `
Object {
  "errors": null,
  "loading": false,
  "message": null,
  "status": "",
  "user": undefined,
}
`;

exports[`The Registration Component should render correctly 1`] = `
<div>
  <SignupForm
    handleClearForm={[Function]}
    handleErrors={[Function]}
    handleFormInput={[Function]}
    handleFormSubmit={[Function]}
    handleSuccess={[Function]}
    showloader={null}
    user={
      Object {
        "confirmedPassword": "",
        "email": "",
        "firstName": "",
        "lastName": "",
        "password": "",
      }
    }
  />
</div>
`;
